Link error

I'm trying to build crate with xargo as noted in readme:
```
curl -L https://raw.githubusercontent.com/japaric/steed/master/Xargo.std.toml > Xargo.toml
curl -LO https://raw.githubusercontent.com/japaric/steed/master/docker/x86_64-unknown-linux-steed.json
xargo build --target x86_64-unknown-linux-steed
```
The following error occurs:
```
ld.lld: error: undefined symbol: __rust_allocate_zeroed
>>> referenced by heap.rs:74 (/home/usov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc/heap.rs:74)
>>>               resolve-240a1ff8cdd5c4fd.0.o:(alloc::heap::allocate_zeroed::hd2157b4681ba8fda) in archive /home/usov/src/resolve-test/target/x86_64-unknown-linux-steed/debug/deps/libresolve-240a1ff8cdd5c4fd.rlib
          
ld.lld: error: undefined symbol: __rust_allocate_zeroed
>>> referenced by heap.rs:74 (/home/usov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc/heap.rs:74)
>>>               idna-be6e683175e98a5b.0.o:(alloc::heap::allocate_zeroed::hd2157b4681ba8fda) in archive /home/usov/src/resolve-test/target/x86_64-unknown-linux-steed/debug/deps/libidna-be6e683175e98a5b.rlib
```
Almost the same with `xargo rustc --target x86_64-unknown-linux-steed -- -C linker=gcc -Z linker-flavor=gcc`:
```
/home/usov/src/resolve-test/target/x86_64-unknown-linux-steed/debug/deps/libresolve-240a1ff8cdd5c4fd.rlib(resolve-240a1ff8cdd5c4fd.0.o): In function `alloc::heap::allocate_zeroed':
/home/usov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc/heap.rs:74: undefined reference to `__rust_allocate_zeroed'
/home/usov/src/resolve-test/target/x86_64-unknown-linux-steed/debug/deps/libidna-be6e683175e98a5b.rlib(idna-be6e683175e98a5b.0.o): In function `alloc::heap::allocate_zeroed':
/home/usov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc/heap.rs:74: undefined reference to `__rust_allocate_zeroed'
collect2: error: ld returned 1 exit status
```

--- a/naive_ralloc/src/lib.rs
+++ b/naive_ralloc/src/lib.rs
@@ -38,6 +38,16 @@ pub extern fn __rust_allocate(size: usize, _align: usize) -> *mut u8 {
 
 #[linkage = "external"]
 #[no_mangle]
+pub extern fn __rust_allocate_zeroed(size: usize, align: usize) -> *mut u8 {
+    unsafe {
+        let result = __rust_allocate(size, align);
+        intrinsics::write_bytes(result, 0, size);
+        result
+    }
+}
+
+#[linkage = "external"]
+#[no_mangle]
 pub extern fn __rust_deallocate(_ptr: *mut u8, _old_size: usize, _align: usize) {
 }
 

